base/applications/mscutils/eventvwr/eventvwr.c

```diff
@@ -4007,11 +4011,79 @@ InitPropertiesDlg(HWND hDlg, PEVENTLOG EventLog)
     }
 }
 
+static
+VOID
+SavePropertiesDlg(HWND hDlg, PEVENTLOG EventLog)
+{
+    LPWSTR lpLogName = EventLog->LogName;
+
+    DWORD dwMaxSize = 0, dwRetention = 0;
+    HKEY hLogKey;
+    WCHAR *KeyPath;    
+    SIZE_T cbKeyPath;
+
+    if (!EventLog->Permanent)
+        return;
+
+    cbKeyPath = (wcslen(EVENTLOG_BASE_KEY) + wcslen(lpLogName) + 1) * sizeof(WCHAR);
+    KeyPath = HeapAlloc(GetProcessHeap(), 0, cbKeyPath);
+    if (!KeyPath)
+    {
+        ShowWin32Error(ERROR_NOT_ENOUGH_MEMORY);
+        return;
+    }
+
+    StringCbCopyW(KeyPath, cbKeyPath, EVENTLOG_BASE_KEY);
+    StringCbCatW(KeyPath, cbKeyPath, lpLogName);
+
+    /* Try to create/open the Event Viewer user key */
+    if (RegCreateKeyExW(HKEY_LOCAL_MACHINE,
+                        KeyPath,
+                        0,
+                        NULL,
+                        REG_OPTION_NON_VOLATILE,
+                        KEY_READ | KEY_WRITE,
+                        NULL,
+                        &hLogKey,
+                        NULL) != ERROR_SUCCESS)
+    {
+        ShowWin32Error(GetLastError());
+        return;
+    }
+
+    HeapFree(GetProcessHeap(), 0, KeyPath);
```

Comment:
```suggestion
    Result = RegOpenKeyExW(hkMachine, KeyPath, 0, KEY_SET_VALUE, &hLogKey);
    HeapFree(GetProcessHeap(), 0, KeyPath);
    if (Result != ERROR_SUCCESS)
    {
        ShowWin32Error(Result);
        return;
    }
```
When the event log key does not exist, it is not recreated by the event log viewer.